src/morale.h

```diff
@@ -76,6 +73,61 @@ enum morale_type : int {
     NUM_MORALE_TYPES
 };
 
+// Morale multiplier
```

Comment:
Yes, I definitely think so, but currently it's shared by both `player` and `morale_point`, so it requires a bit awkward shuffling among .cpp files. Though it could be done nicely if we put all morale code in a separate class with narrow interface so that nobody outside morale.cpp knows about neither `morale_point` nor `morale_mult`. It's something that IMO worth doing since the player is too god-like. Then the player could simply aggregate it:

```
class morale {
    public:
        morale( player u );
        int get_level();
        void add(..);
        void remove(..);
        int has(..);
        ...
    private:
        player p; // Back pointer
        std::vector<morale_point> morale;
        mutable int level;
        mutable bool level_is_valid;
        /** Returns current traits multiplier */
        morale_mult get_traits_mult() const;
        /** Returns current effects multiplier */
        morale_mult get_effects_mult() const;        
        ...
}
```
That's my intention for further morale PRs.